app/controllers/api/apartments_controller.rb

```diff
@@ -0,0 +1,66 @@
+module Api
+  class ApartmentsController < Api::ApplicationController
+    def index
+      apartments = Apartment.all
+      render json: { apartments: }
+    end
+
+    def show
+      apartment = Apartment.find(params[:id])
+      render json: { apartment: }
+    end
+
+    def create
+      if current_user.role != 'admin'
+        render(json: { error: 'Only admin can create an apartment' }, status: 401)
+        return
+      end
+
+      apartment = Apartment.new(name: apartment_params[:name],
+                                description: apartment_params[:description],
+                                photo: apartment_params[:photo],
+                                city: apartment_params[:city],
+                                price: apartment_params[:price])
+
+      if apartment.save
+        render(json: { success: 'Apartment was created', apartment: { id: apartment.id } }, status: 200)
+      else
+        render(json: { error: apartment.errors.full_messages }, status: 422)
+      end
+    end
+
+    def update
+      if current_user.role != 'admin'
+        render(json: { error: 'Only admin can update an apartment' }, status: 401)
+        return
+      end
+
+      apartment = Apartment.find(params[:id])
+
+      if apartment.update(apartment_params)
+        render(json: { success: 'Apartment was updated' }, status: :ok)
+      else
+        render(json: { error: apartment.errors.full_messages }, status: 422)
+      end
+    end
+
+    def destroy
+      if current_user.role != 'admin'
+        render(json: { error: 'Only admin can destroy an apartment' }, status: 401)
+        return
+      end
+
+      apartment = Apartment.find(params[:id])
+      apartment.reservations.destroy_all
```

Comment:
[SUGGESTION]: Kindly remove this line, as it is not needed. In the Apartment model you will see that:

```ruby
has_many :reservations, dependent: :destroy
```

so it already destroys all the dependent reservations when the apartment is destroyed.